### DISCUSSION TITLE
How to connect from PHP PDO to Neon? - SNI issues

### DISCUSSION THREAD

**Discussion by tomekit** [neondatabase/neon#3845] [Q&A]
I am trying to connect to Neon from PHP8 with PDO (https://www.php.net/manual/en/ref.pdo-pgsql.php), but I am facing SNI issues, that I can't resolve. It's purely PHP related issue, as I've managed to connect to NeonDB from Go without problems.

My initial DSN:
```
pgsql:dbname=%s;host=%s;port=%s;sslmode=require
```

returns SNI error:
```
 ERROR:  Endpoint ID is not specified. Either please upgrade the postgres client library (libpq) for SNI support or pass the endpoint ID (first part of the domain name) as a parameter: '?options=project%3D<endpoint-id>'. See more at https://neon.tech/sni
```

Appending: `;options=project%%3Dproject-name-022114` (bear in mind second % escape, as I am using `sprintf`) doesn't seem to work. It doesn't seem that `options` is supported in the PHP DSN: https://www.php.net/manual/en/ref.pdo-pgsql.connection.php

Passing the `options` to PDO constructor doesn't seem to work either:
```
$options = [
	\PDO::ATTR_ERRMODE => \PDO::ERRMODE_EXCEPTION,
	'project' => "project-name-022114",
];

static::$pdo = new \PDO($dsn, $dbConfig['username'], $dbConfig['password'], $options);
```

Not entirely sure what would be the best way to overcome SNI issues using PHP PDO.

In principle I could follow approach D): https://neon.tech/docs/connect/connectivity-issues#d-specify-the-endpoint-id-in-the-password-field 
however I don't feel comfortable downgrading password hashing method.

Do you have any experience with PHP and have an idea how to solve this?

**Comment by StreetXHasu** [comment]
`DB_DATABASE="main options=project=<project>"` 
I also tried to use this construction and pass this option with the database, but nothing happened

laravel 10

**Comment by danieltprice** [comment]
Hi, PDO uses libpq, and SNI support is included in libpq in v14 or higher. Try checking your libpq version and upgrading if it's lower than 14.
